clear-site-data/storage.https.html

```diff
@@ -22,6 +24,28 @@
       test(function() {}, "Populate backends.");
 
       TestUtils.populateStorage()
+          .then(() => {
+            return navigator.serviceWorker.getRegistration("support/").then(function(reg) {
+              return TestUtils.waitForServiceWorkerActive(reg);
+            });
+          })
+          .then(() => {
+            const serviceWorkerResponseOk = TestUtils.getServiceWorkerResponseOk();
+
+            // Create iFrame in the service worker's scope. This page will make a request
+            // for another page that is only served by the service worker
+            serviceWorkerTestPageIFrame = document.createElement("iframe");
+            serviceWorkerTestPageIFrame.src = "support/page_using_service_worker.html";
+            document.body.appendChild(serviceWorkerTestPageIFrame);
+
+            promise_test(function() {
```

Comment:
Can we directly do:

```
const resp = await serviceWorkerTestPageIFrame.contentWindow.fetch('sw-page');
assert_true(resp.ok);
```

Then you don't need the messaging.

Reply:
@mattto I don't think so. From what I can tell, doing `serviceWorkerTestPageIFrame.contentWindow.fetch` still uses the controller of the test runner page. When using a relative url for `serviceWorkerTestPageIFrame.contentWindow.fetch`, the path must be relative to the test page, which is another hint that such an approach isn't going to work.

Reply:
That a bit surprising, it's fairly common for WPT service worker tests to do this: https://cs.chromium.org/search/?q=contentwindow.fetch+file:%5Esrc/third_party/blink/web_tests/external/wpt/service-workers/service-worker/+package:%5Echromium$&type=cs

This test looks similar to https://cs.chromium.org/chromium/src/third_party/blink/web_tests/external/wpt/service-workers/service-worker/unregister-controller.https.html?l=67&rcl=23d09c4f8e7f7195f04f0754d3070f7c9de49408 (although that's using XHR for some reason)

Reply:
Hmmm, ok let me dig deeper to see if I'm missing something

Reply:
@mattto ah you're right. I've updated the test to use `.contentWindow` instead of `postMessage`. Much simpler/nicer now, thanks.